### PR TITLE
Define a `RELEASE.MD` for the Alertmanager Project

### PR DESCRIPTION
**Pull request by gotjosh**
Will helps us get to a more regular cadance of releases.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Releases
+This page describes the release process and the currently planned schedule for upcoming releases as well as the respective release shepherd. Release shepherds are chosen on a voluntary basis.
+
+## Release Schedule
+
+Release cadence of first pre-releases being cut is 6 weeks.
```
Comment (simonpasquier) [inline RELEASE.md:6]: I thought that we would start with a 12 weeks/3 months cadence?

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| release series | date (year-month-day) | release shepherd              |
+|----------------|-----------------------|-------------------------------|
+| v0.26          | 2023-05-31            | Josh Abreu (Github: @gotjosh) |
+| v0.27          | 2023-07-15            | Josh Abreu (Github: @gotjosh) |
+
+If you are interested in volunteering please create a pull request against the [prometheus/alertmanager](https://github.com/prometheus/alertmanager) repository and propose yourself for the release of your choice.
+
+## Release shepherd responsibilities
+
+The release shepherd is responsible for the entire release series of a minor release.
```
Comment (simonpasquier) [inline RELEASE.md:17]: Do you mean to say that the release shepherd should actively look after all pull requests during the 6 weeks period?

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+* On the date listed in the table above, the release shepherd cuts the first pre-release (using the suffix `-rc.0`) and creates a new branch called  `release-<major>.<minor>` starting at the commit tagged for the pre-release. In general, a pre-release is considered a release candidate (that's what `rc` stands for) and should therefore not contain any known bugs that are planned to be fixed in the final release.
+* With the pre-release, the release shepherd is responsible for making sure the release candidate is in a good state.  After the pre-release is out for 3 days it is promoted to a stable release.
+* If regressions or critical bugs are detected, they need to get fixed before cutting a new pre-release (called `-rc.1`, `-rc.2`, etc.).
+
+See the next section for details on cutting an individual release.
+
+## Release Instructions (TBD)
```
Comment (simonpasquier) [inline RELEASE.md]: I think that we can link to the Prometheus RELEASE.md file and only highlight the differences.

**Comment by gotjosh** [comment]
@simonpasquier let's move the discussion to the next PR, github is acting up on me.

**Pull request closed without merging**